shipwright/build.py

```diff
@@ -66,20 +67,26 @@ def process_event_(evt):
     return (process_event_(evt) for evt in build_evts)
 
 
-@fn.composed(maybe(fn._0), fn.search(r'^Successfully built ([a-f0-9]+)\s*$'))
 def success(line):
     """
     >>> success('Blah')
     >>> success('Successfully built 1234\\n')
     '1234'
     """
+    match = RE_SUCCESS.search(line)
+    if match:
+        return match.group(1)
+    return None
 
 
-@fn.composed(fn.first, fn.filter(None), fn.map(success))
 def success_from_stream(stream):
     """
-
     >>> stream = iter(('Blah', 'Successfully built 1234\\n'))
     >>> success_from_stream(stream)
     '1234'
     """
+    for x in stream:
+        result = success(x)
+        if result:
+            return result
+    return None
```

Comment:
I believe the API here is that the `None` is meaningful, so I think it's important to return it explicitly.